indico/modules/rb_new/controllers/backend.py

```diff
@@ -298,26 +298,38 @@ def _process(self):
 
 
 class RHCreateRoomBlocking(RHRoomBookingBase):
-    @use_kwargs({
+    @use_args({
         'room_ids': fields.List(fields.Int(), missing=[]),
         'start_date': fields.Date(),
         'end_date': fields.Date(),
         'reason': fields.Str(),
-        'allowed_principals': fields.List(fields.Dict(), missing=[])
+        'allowed_principals': fields.List(fields.Dict())
     })
-    def _process(self, room_ids, start_date, end_date, reason, allowed_principals):
-        rooms = Room.query.filter(Room.id.in_(room_ids)).all()
-        allowed = []
-        for obj in allowed_principals:
-            if obj.get('is_group'):
-                allowed.append(GroupProxy(obj['id'], provider=obj['provider']))
-            else:
-                allowed.append(User.query.filter(User.id == obj['id']).one())
-        blocking = create_blocking(rooms, start_date, end_date, reason, allowed)
+    def _process(self, args):
+        blocking = create_blocking(args)
```

Comment:
I know we do it in many other places, but I think passing/accepting an `args` dict in an operation function is not great - I'd rather have the operation take explicit arguments and then unpack `**args` here.

When just accepting a dict of "args" in the operation you need to go into the modal again to see what the operation actually accepts...

(nothing that needs to be changed now, just something that came to my mind a few times now when seeing operation functions)